core/story-api/src/main/java/org/lgna/story/implementation/DoubleProperty.java

```diff
@@ -43,16 +43,41 @@
 
 package org.lgna.story.implementation;
 
+import edu.cmu.cs.dennisc.animation.Style;
+import edu.cmu.cs.dennisc.java.util.logging.Logger;
+
 /**
  * @author Dennis Cosgrove
  */
 public abstract class DoubleProperty extends Property<Double> {
-  public DoubleProperty(PropertyOwnerImp owner) {
+  public static Double CLOSE_TO_ZERO = .01;
+
+  private Double minValue;
+
+  public DoubleProperty(PropertyOwnerImp owner, Double minValue) {
     super(owner, Double.class);
+
+    this.minValue = minValue;
+  }
+
+  public DoubleProperty(PropertyOwnerImp owner) {
+    this(owner, Double.NaN);
   }
 
   @Override
   protected Double interpolate(Double a, Double b, double portion) {
     return a + ((b - a) * portion);
   }
+
+  @Override
+  public void animateValue(final Double value, double duration, Style style) {
+    Double newValue = value;
+
+    if (!minValue.isNaN() && value < minValue) {
+      Logger.outln("Attempt to set double property below the minimum of " + minValue + " to " + value + " ignored.");
```

Comment:
I would be lying if I said I hate to be pedantic, so here is my pedantry.
The value is not ignored, because it will animate something, just stopping at a value that makes sense.
Maybe, `…+ minValue + " to " + value + " will stop at the minimum."`? Kind of awkward.
Or just ignore me, since this is only a minor bit of logging.